nav2_world_model/include/nav2_world_model/costmap_representation.hpp

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_WORLD_MODEL__COSTMAP_REPRESENTATION_HPP_
+#define NAV2_WORLD_MODEL__COSTMAP_REPRESENTATION_HPP_
+
+#include <string>
+#include <memory>
+#include <vector>
+
+#include "rclcpp/rclcpp.hpp"
+#include "nav2_world_model/world_representation.hpp"
+#include "nav2_costmap_2d/costmap_2d_ros.hpp"
+#include "tf2_ros/transform_listener.h"
+#include "nav2_util/point.hpp"
+#include "nav2_world_model/region_visualizer.hpp"
+
+namespace nav2_world_model
+{
+
+class CostmapRepresentation : public WorldRepresentation
+{
+public:
+  CostmapRepresentation(
+    const std::string name,
+    rclcpp::Node::SharedPtr & node,
+    rclcpp::executor::Executor & executor,
+    rclcpp::Clock::SharedPtr & clock);
+
+  GetCostmap::Response getCostmap(const GetCostmap::Request & request) override;
```

Comment:
Might want to think of a better API name than `getCostmap` as sometimes people might not be using costmaps (ie elevation maps, non-regular-grid-maps, etc)

Reply:
Certainly, different world model clients migh require to obtain different expressions of the world. I think we will be improving this API as we go.

However, we should point out that the internal world representation used should be independent of the interface. If world model has a client, navfn for instance, that requires a costmap, the world representation should support the conversion.

Reply:
I think the API needs to be thought through with some of those other representations in mind. Is there a use case for using a elevation map as the internal representation, but then converting it to a costmap? If so, should it be the responsibility of the elevation map to make that conversion? Or should the representations provide their data, with the conversion being the responsibility of another component.

Reply:
All of these are valid points and have been included in #565.

This PR is a minor refactor of the world model, #565 will be a redesign of the world model. For now, map server and global planner use cost maps so no need to extend the API.